src/components/SearchBox/SearchBox.tsx

```diff
@@ -106,13 +107,25 @@ export class SearchBox extends React.PureComponent<SearchBoxProps, SearchBoxStat
       return;
     }
 
-    this.setState({
-      term: q,
-    });
+    this.setState(
+      {
+        term: q,
+      },
+      this.debounce(
+        () =>
+          this.props.search.search(this.state.term).then(res => {
+            this.setResults(res, this.state.term);
+          }),
+        300,
+      ),
+    );
+  };
 
-    this.props.search.search(event.target.value).then(res => {
-      this.setResults(res, q);
-    });
+  debounce = (fn: (..._) => any, wait: number = 1): (() => void) => {
```

Comment:
Could you remove your hand-made debounce and use `debounce` from `decko` (we already use this dependency) instead?

Reply:
I tried using the `debounce` implementation from decko as such - 
```javascript
@debounce
search = (event: React.ChangeEvent<HTMLInputElement>) => {
...
}
```
But this showed the following warning when trying to compile- 
`TS1240: The return type of a property decorator function must be either 'void' or 'any'. `

Furthermore, when I try to use `delay` as argument as such - 
```javascript
@debounce(300)
search = (event: React.ChangeEvent<HTMLInputElement>) => {
...
}
```
I get the following warning - 
`TS1240: 
Unable to resolve signature of property decorator when called as an expression.`

As the script was not compiling, I opted to use a hand-made debounce. One thing to note is that the files in which the `debounce` from `decko` is already being used (e.g. `ScrollService.ts` and `HistoryService.ts` are `ts` file, whereas `SearchBox` is a `tsx` file. Maybe there needs to be a change in lint configuration for `tsx` files?

Reply:
Use 

```
  @bind
  @debounce
  search(event: React.ChangeEvent<HTMLInputElement>) {
```

Note `search` is not field with lambda function as value but method.

Reply:
Pushed the changes